test/scripts/transforms/test_loq_transforms.py

```diff
@@ -0,0 +1,213 @@
+save_sector_reduction(output_workspaces, "_horizontal_sector")
+
+ici.SetPhiLimit(60, 120, use_mirror=True)
+output_workspaces=ici.WavRangeReduction(None, None, ici.DefaultTrans, combineDet="merged")
+save_sector_reduction(output_workspaces, "_vertical_sector")
+
+# Now perform the overlap reduction
+cleanup_and_setup_ici()
+ici.Set1D()
+for i in range(len(default_slice_wavs) - 1):
+  output_workspace = ici.WavRangeReduction(default_slice_wavs[i], default_slice_wavs[i + 1], False, combineDet='merged')
+  SaveNXcanSAS(**get_nxcansas_kwargs(output_workspace, ws_suffix=f"{default_slice_wavs[0]}_{default_slice_wavs[-1]}"))
+  output.append(f"{output_workspace}_auto.h5")
+"""
+    )
+
+
+@pytest.fixture()
+def reduction():
+    """
+    Reduction fixture
+    :return:
+    """
+    mock = Mock()
+    mock.inputs = {
+        "user_file": "/extras/loq/BestMaskFile.toml",
+        "run_number": 10,
+        "scatter_transmission": 9,
+        "scatter_direct": 3,
+        "can_scatter": 5,
+        "can_transmission": 4,
+        "can_direct": 3,
+        "sample_thickness": 2.0,
+        "sample_geometry": "Disc",
+        "sample_height": 8.0,
+        "sample_width": 8.0,
+    }
+    return mock
+
+
+def test_loq_transform_apply(script, reduction):  # noqa: PLR0912, C901
+    """
+    Test loq transform applies correct updates to script
+    :param script: The script fixture
+    :param reduction: The reduction fixture
+    :return: None
+    """
+    transform = LoqTransform()
+
+    original_lines = script.value.splitlines()
+    transform.apply(script, reduction)
+    updated_lines = script.value.splitlines()
+    assert len(original_lines) == len(updated_lines)
+    for index, line in enumerate(updated_lines):
+        if line.startswith("user_file"):
+            assert line == 'user_file = "/extras/loq/BestMaskFile.toml"'
+        elif line.startswith("sample_scatter"):
+            assert line == "sample_scatter = 10"
+        elif line.startswith("sample_transmission"):
+            assert line == "sample_transmission = 9"
+        elif line.startswith("sample_direct"):
+            assert line == ("sample_direct = 3")
+        elif line.startswith("can_scatter"):
+            assert line == "can_scatter = 5"
+        elif line.startswith("can_transmission"):
+            assert line == "can_transmission = 4"
+        elif line.startswith("can_direct"):
+            assert line == "can_direct = 3"
+        elif line.startswith("sample_thickness"):
+            assert line == "sample_thickness = 2.0"
+        elif line.startswith("sample_geometry"):
+            assert line == 'sample_geometry = "Disc"'
+        elif line.startswith("sample_height"):
+            assert line == "sample_height = 8.0"
+        elif line.startswith("sample_width"):
+            assert line == "sample_width = 8.0"
+        else:
+            assert line == original_lines[index]
```

Comment:
Could move the lines for the script being checked into a dictionary. Easier to visualise and only need one assert:
```suggestion
    replacements = {
        "user_file": 'user_file = "/extras/loq/BestMaskFile.toml"',
        "sample_scatter": "sample_scatter = 10",
        "sample_transmission": "sample_transmission = 9",
        "sample_direct": "sample_direct = 3",
        "can_scatter": "can_scatter = 5",
        "can_transmission": "can_transmission = 4",
        "can_direct": "can_direct = 3",
        "sample_thickness": "sample_thickness = 2.0",
        "sample_geometry": 'sample_geometry = "Disc"',
        "sample_height": "sample_height = 8.0",
        "sample_width": "sample_width = 8.0",
    }
    
    for index, line in enumerate(updated_lines):
        for key, expected_line in replacements.items():
            if line.startswith(key):
                assert line == expected_line 
                break
        else:
            assert line == original_lines[index]
```

Reply:
Fundamentally the above suggestion has performance characteristics of O(n^2), vs the original at O(n), I inherently don't program like that but in this scenario, I think it's a little clearer and it's a defined length anyways.